File: MSRpdf_cop_Frank/MSRpdf_cop_Frank.m

function MSRpdf_cop_Frank
[u,v] = meshgrid(0:0.02:1);
p     = 2;

gcopuly = -p.*(exp(-p.*(u+v+1)) - exp(-p.*(u+v)))./...
    ((exp(-p) + exp(-p.*(u+v)) - exp(-p.*u) - exp(-p.*v)).^2);
grid on

mesh(u, v, gcopuly)
xlabel('X');
ylabel('Y');
zlabel('Z');
title('Frank Copula Density, \\theta = 2')

File: MSRpdf_cop_Frank/MSRpdf_cop_Frank.R
rm(list = ls(all = TRUE))
#setwd("C:/...")


#install.packages("lattice")
library(lattice)

u = seq(0, 1, by = 0.02)
v = u
m = length(u) 
n = length(v)
U = matrix(rep(u, each = n), nrow = n)
V = matrix(rep(v, m), nrow = n)

p = 2

gcopuly = -p*(exp(-p*(U+V+1)) - exp(-p*(U+V)))/ ((exp(-p) + exp(-p*(U+V)) - exp(-p*U) - exp(-p*V))^2)

s = expand.grid(u = u, v = v)
wireframe(gcopuly ~ u*v, s, shade = TRUE, xlab = "X", ylab = "Y", zlab = "Z", 
    main = "Frank Copula Density, \\theta = 2", scales = list(arrows = FALSE))